user/newlib/overrides/include/sys/signal.h

```diff
@@ -0,0 +1,49 @@
+#ifndef _SYS_SIGNAL_H_
+#define _SYS_SIGNAL_H_
+
+#include <unistd.h>
+
+/* TODO: Should we somehow merge this file with kernel's signal.h? */
```

Comment:
Yes. Use `_KERNELSPACE` preprocessor definition to limit kernel interface visibility for userspace programs.